astro-migration-provider/astronomer/migration/operators.py

```diff
@@ -0,0 +1,193 @@
+import os
+import requests
+
+from airflow.models import BaseOperator, Connection, Variable, Pool
+
+from airflow.utils.session import provide_session
+from python_graphql_client import GraphqlClient
+from sqlalchemy.orm import Session
+from typing import Any, Sequence
+
+class AstroVariableMigrationOperator(BaseOperator):
```

Comment:
IMO this should be one operator - `AstroMigrationOperator` so that the DAG that needs to implement it is as simple as possible, and relies on sensible defaults as much as possible

Reply:
What if customers don't want to migrate connections, but want to migrate Variables? Keeping things atomic may also be easier to maintain.

Reply:
I think we chatted about this a bit in the call yesterday - The operator should take configuration which can allow or disallow migrating certain things, but the highest priority will be that it "just works" and has all batteries included. We should shoot for simplicity over covering every case - as this is a fallback anyway, and is hopefully a small minority of cases.

Maybe these individual operators - but then a single operator that does everything via calling the other operators' `.execute` methods? and that's what we recommend people use?

Reply:
That feels more gross to me from a maintainability perspective -- and if our readme already is providing them with the DAG that they're deploying, why does it matter if we have 3 operators vs. 1?

Reply:
followed up in slack - recording here. We discussed the maintainability aspects and that it doesn't necessarily matter to finish out the Spike, but that having a single operator that encompasses the others that do one thing could be a "best of both worlds" option

Reply:
Pushed resolve.